### ISSUE TITLE
Width calculation in column's `setAutoSize()` with wide-chars

### ISSUE THREAD

**Issue by boxsnake** [PHPOffice/PhpSpreadsheet#443]
This is:
- [x] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)

### What is the expected behavior?
Each wide char should be ~2 normal-char-wide when using `setAutoSize(true)`, e.g.
![image](https://user-images.githubusercontent.com/3257783/38072589-9e94f63c-3359-11e8-9d4c-8f061019fee8.png)

### What is the current behavior?
Each wide char only takes one-char width when using `setAutoSize(true)`.
![image](https://user-images.githubusercontent.com/3257783/38072660-f42c5572-3359-11e8-8e85-85bcc4e2c625.png)

### What are the steps to reproduce?

Please provide a [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve) of code that exhibits the issue without relying on an external Excel file or a web server:

```php
<?php
require __DIR__ . '/vendor/autoload.php';

use PhpOffice\PhpSpreadsheet\Spreadsheet;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx as XlsxWriter;

// Create new Spreadsheet object
$spreadsheet = new Spreadsheet();
$sheet       = $spreadsheet->getActiveSheet();

// add code that show the issue here...
// ... Get Data called `$data` (2D-array), each row has index `is_synced`
foreach($data as $i => $item) {
    $rowNum = $i + 1;
    $sheet->setCellValueExplicit("D{$rowNumber}", !empty($row['is_synced']) ? '已同步' : '未同步', \PhpOffice\PhpSpreadsheet\Cell\DataType::TYPE_STRING);
}

$writer = new XlsxWriter($spreadsheet);
$writer->save('php://output');

# Free Memory
$spreadsheet->disconnectWorksheets();
unset($spreadsheet);
exit;
```

### Which versions of PhpSpreadsheet and PHP are affected?
PHP 7.0.12 (NTS)

**Comment by boxsnake** [comment]
I have a workaround hack for it, but I do not want to fork and make PR.

For `\PhpOffice\PhpSpreadsheet\Shared\StringHelper` class, should have a function like:
```php
public static function monoWidth($value)
{
    return mb_strlen($value, 'GB2312');
}
# Or
public static function monoWidth($value)
{
    return mb_strlen($value, 'GB18030');
}
```

And use `StringHelper::monoWidth()` in `\PhpOffice\PhpSpreadsheet\Shared\Font::getTextWidthPixelsApprox()` instead.

**Comment by boxsnake** [comment]
I made a simple workaround for font Calibri, but I eagerly hope you developers can fix this.

```php
class ...
{
    private function autoSize(\PhpOffice\PhpSpreadsheet\Worksheet\Worksheet &$sheet)
    {
        $list    = $sheet->rangeToArray('A1:' . $sheet->getHighestColumn() . $sheet->getHighestRow(), '', TRUE, TRUE, TRUE);
        $colDims = [];

        foreach($list as $rowIndex => $rowData) {
            foreach($rowData as $colIndex => $colData) {
                $charWidth = \App\charlen($colData);

                if(empty($colDims[$colIndex]))
                    $colDims[$colIndex] = $charWidth;
                elseif(!empty($colDims[$colIndex]) && $charWidth > $colDims[$colIndex])
                    $colDims[$colIndex] = $charWidth;
            }
        }

        foreach($colDims as $colIndex => $charWidth) {
            $sheet->getColumnDimension($colIndex)->setWidth($charWidth + 3); // padding: 3-char width
        }
    }
    ...
}
```

**Comment by oleibman** [comment]
Fixed by PR #3416.
